tests/NuGet.IndexingTests/TestSupport/MockSearcherManager.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using NuGet.Indexing;
+
+namespace NuGet.IndexingTests.TestSupport
+{
+    public class MockSearcherManager : NuGetSearcherManager
+    {
+        private string _indexName;
+
+        public MockSearcherManager(string indexName) : base(indexName, new MockLogger(), null, null, null)
```

Comment:
I think given that it's just a small constructor that would be calling another constructor I don't see an issue with touching the code. I've had to add constructors for testing before--sometimes there's no other way to do it due to how the language works.